src/vopr.zig

```diff
@@ -690,7 +690,13 @@ pub const Simulator = struct {
                 assert(simulator.cluster.replica_health[replica.replica] == .up);
                 const commit_max = @min(replica.op, cluster_commit_max);
                 const commit_min = replica.commit_min;
-                if (commit_min < commit_max) {
+
+                // WAL repair can only be performed by replicas in normal state or potential
+                // primaries (see repairs_allowed in replica.zig). Use the difference between view
+                // and log_view as a heuristic for whether a replica has attempted repair or not.
+                if (commit_min < commit_max and
+                    replica.view - replica.log_view <= simulator.cluster.replica_count)
+                {
```

Comment:
Rather than using a heuristic, why not compare the replica's `op_checkpoint` with the cluster's `op_checkpoint`?

Reply:
Mhmmmm good point, we could use `op_checkpoint` to figure this out as well!

However, I want to press pause on this PR till our discussion tomorrow, as it triggered an entire train of thought that I wanted to discuss with you! It is related to lagging replicas also initiating `repair` during view change, so they can commit as much as they can (and even step up as potential primaries if need be). 😄